src/video/sdl2_v.cpp

```diff
@@ -835,6 +822,29 @@ void VideoDriver_SDL::LoopOnce()
 			Paint();
 		}
 	}
+
+	/* If we are not in fast-forward, create some time between calls to ease up CPU usage. */
+	if (!_fast_forward || _pause_mode) {
+		/* See how much time there is till we have to process the next event, and try to hit that as close as possible. */
+		uint next_tick = next_draw_tick < next_game_tick ? next_draw_tick : next_game_tick;
+		uint next_delta = next_tick - SDL_GetTicks();
+
+		/* If we missed our deadline already (read: GameLoop took longer than 30ms), don't sleep. */
+		if (next_delta > 0) {
+			/* Release the thread while sleeping */
+			if (_draw_mutex != nullptr) {
+				draw_lock.unlock();
```

Comment:
std::condition_variable::wait_until / wait_for combines the waiting with the unlock+lock.
Maybe it performs better?

Reply:
The internet associates condition_variable's timeouts mostly with system_clock. So it's probably not precise enough.